source/class/cv/io/mqtt/Client.js

```diff
@@ -0,0 +1,263 @@
+        console.log( 'Connection Lost: ' + responseObject.errorMessage, responseObject );
+        self.setConnected(false);
+      };
+
+      this.client.onMessageArrived = function (message) {
+        let update = {};
+        update[ message.topic ] = message.payloadString;
+        self.update(update);
+      };
+
+      this.client.connect( options );
+    },
+
+    /**
+     * Authorize a Request by adding the necessary headers.
+     * @param req {qx.io.request.Xhr}
+     */
+    authorize: function (req) {},
+
+    /**
+     * return the relative path to a resource on the currently used backend
+     *
+     * @param name {String} Name of the resource (e.g. login, read, write, rrd)
+     * @param params {Map?} optional data needed to generate the resource path
+     * @return {String|null} relative path to the resource, returns `null` when the backend does not provide that resource
+     */
+    getResourcePath : function (name, params) {},
+
+    /**
+     * This client provides an own processor for charts data
+     * @return {Boolean}
+     */
+    hasCustomChartsDataProcessor : function () {},
+
+    /**
+     * For custom backend charts data some processing might be done to convert it in a format the CometVisu can handle
+     * @param data {var}
+     */
+    processChartsData : function (data) {},
+
+    /**
+     * This function sends a value
+     * @param address {String} address to send the value to
+     * @param value {String} value to send
+     * @param options {Object} optional options, depending on backend
+     *
+     */
+    write : function (address, value, options) {
+      if( this.isConnected() ) {
+        let message = new Paho.MQTT.Message( value );
+        message.destinationName = address;
+        message.qos = options.qos;
+        message.retained = options.retain;
+        this.client.send(message);
+      }
+    },
+
+    /**
+     * Get the last recorded error
+     *
+     * @return {{code: (*|Integer), text: (*|String), response: (*|String|null), url: (*|String), time: number}|*}
+     */
+    getLastError: function() {},
+
+    /**
+     * Restart the connection
+     */
+    restart: function(full) {},
+
+    /**
+     * Handle the incoming state updates. This method is not implemented by the client itself.
+     * It is injected by the project using the client.
+     * @param json
+     */
+    update: function(json) {}, // jshint ignore:line
+
+    /**
+     * Can be overridden to record client communication with backend
+     * @param type {String} type of event to record
+     * @param data {Object} data to record
+     */
+    record: function(type, data) {},
+
+    /**
+     * Can be overridden to provide an error handler for client errors
+     * @param type {Number} one of cv.io.Client.ERROR_CODES
+     * @param message {String} detailed error message
+     * @param args
+     */
+    showError: function(type, message, args) {}
+  }
+
```

Comment:
Please add a constructor to clear local objects (at least the _client):
```js
 destruct: function () {
   this,_client = null;
 }
 ```
 
 Although the backend client is a singleton that will not be destroyed during the livetime of the current cometvisu instance, we should "just do this" to be clean.